components/arm/fake/fake.go

```diff
@@ -125,6 +131,22 @@ type Arm struct {
 	model      referenceframe.Model
 }
 
+// UpdateAction helps hinting the reconfiguration process on what strategy to use given a modified config.
+// See config.UpdateActionType for more information.
+func (a *Arm) UpdateAction(c *config.Component) config.UpdateActionType {
+	var err error
+	if newCfg, ok := c.ConvertedAttributes.(*AttrConfig); ok {
+		armModel := newCfg.ArmModel
+		modelPath := newCfg.ModelFilePath
+		a.model, err = buildModel(armModel, modelPath, a.Name)
```

Comment:
`UpdateAction` only support return value of type `config.UpdateActionType`.
Hence, if we have an error we *cannot* return it.
Proposed solution is to notify the user of the error through a debug statement. 
Open to suggestions.
LMK.

Reply:
This should indeed be an error log. Logging things at an Error level do not in and of themselves have any effect on the running of the server.